src/execution_plan/ops/op_index_scan.c

```diff
@@ -0,0 +1,73 @@
+#include "op_index_scan.h"
+
+OpBase *NewIndexScanOp(Graph *g, Node **node, IndexBounds *index_data) {
+  return (OpBase*)NewIndexScan(g, node, index_data);
+}
+
+IndexScan* NewIndexScan(Graph *g, Node **node, IndexBounds *index_data) {
+
+  IndexScan *indexScan = malloc(sizeof(IndexScan));
+  indexScan->node = node;
+  indexScan->_node = *node;
+
+  indexScan->index = index_data->index;
+  SIValue *upper = NULL;
+  // Only the upper bound must be cloned, as the lower bound is only accessed in the skiplistIterateRange call directly below
+  if (index_data->upper) {
+    upper = malloc(sizeof(SIValue));
+    *upper = SI_Clone(*index_data->upper);
+  }
+
+  if (index_data->iter_type == T_STRING) {
```

Comment:
The code for the IndexScan op is largely the same as what had been present in PR #109, except that the `IndexBounds` struct is now used to construct the IndexIterator.